src/profile-logic/import/simpleperf.js

```diff
@@ -0,0 +1,644 @@
+      let category: IndexIntoCategoryList = Categories.Other;
+      if (filePath === '[kernel.kallsyms]' || filePath.endsWith('.ko')) {
+        category = Categories.Kernel;
+      } else if (filePath.endsWith('.so')) {
+        category = fileInSystem ? Categories.System : Categories.Native;
+      } else if (
+        filePath === '[JIT app cache]' ||
+        filePath.endsWith('.vdex') ||
+        filePath.endsWith('.apk') ||
+        filePath.endsWith('.jar') ||
+        filePath.endsWith('.oat') ||
+        filePath.endsWith('.odex')
+      ) {
+        const isJavaMethod =
+          methodName.startsWith('java.') ||
+          methodName.startsWith('javax.') ||
+          methodName.startsWith('kotlin.') ||
+          methodName.startsWith('kotlinx.') ||
+          methodName.startsWith('dalvik.');
+        const isAndroidMethod =
+          methodName.startsWith('android.') ||
+          methodName.startsWith('com.android.') ||
+          methodName.startsWith('androidx.') ||
+          methodName.startsWith('libcore.');
+        category =
+          fileInSystem || isAndroidMethod || isJavaMethod
+            ? Categories.System
+            : Categories.Java;
+      }
+
+      const frameIndex = this.frameTable.findOrAddFrame(funcIndex, category);
+
+      prefixStackId = this.stackTable.findOrAddStack(
+        frameIndex,
+        prefixStackId,
+        category
+      );
+    }
+
+    this.sampleTable.stack.push(prefixStackId);
+    $NotNull(this.sampleTable.time).push(toMilliseconds(sample.time ?? 0));
+
+    if (this.sampleTable.weight) {
+      const weight =
+        this.cpuClockEventId >= 0 && sample.eventTypeId === this.cpuClockEventId
+          ? toMilliseconds(sample.eventCount ?? 0)
+          : 0;
+      this.sampleTable.weight.push(weight);
+    }
+
+    this.sampleTable.length++;
+  }
+}
+
+class FirefoxProfile {
+  threads: FirefoxThread[] = [];
+  threadMap: Map<number, FirefoxThread> = new Map();
+
+  fileMap: Map<number, report.IFile> = new Map();
+
+  eventTypes: string[];
+  cpuClockEventId: number;
+
+  appPackageName: ?string | null;
+  sampleCount: number = 0;
+  lostCount: number = 0;
+
+  toJson(): SerializableProfile {
+    return {
+      meta: this.getProfileMeta(),
+      libs: [],
+      threads: this.threads.map((thread) => thread.toJson()),
+    };
+  }
+
+  getProfileMeta(): ProfileMeta {
+    return {
+      // The interval at which the threads are sampled.
+      interval: 0,
+      startTime: 0,
+      processType: 0,
+      categories: Categories.toJson(),
+      product: this.appPackageName ?? 'Android Profile',
+      stackwalk: 0,
+      // This is the Gecko profile format version (the unprocessed version received directly
+      // from the browser.)
+      version: 30,
+      // This is the processed profile format version.
+      preprocessedProfileVersion: 50,
+
+      symbolicationNotSupported: true,
+      markerSchema: [],
+
+      // platform: "Android",
+      // device?: string,
```

Comment:
Just wondering if these comments were supposed to be uncommented?

Reply:
Sorry for this. I was playing around. We don't really have "device" information and the `platform` param here seems to make no difference in UI, or at lease I don't know how it is used. Planning to remove, let me know if it should be set to something more sensible.

Reply:
This is how we handle the "platform" information: https://github.com/firefox-devtools/profiler/blob/ae149d7439f6a15e2088a8c20f58474415af1f35/src/profile-logic/profile-metainfo.js#L58-L65

Very imperfect :-) but you can probably leverage this here.

Reply:
This helped. Added both `toolkit` and `platform` to meta.